packages/features/bookings/Booker/Booker.tsx

```diff
@@ -0,0 +1,193 @@
+    eventId: event?.data?.id,
+    rescheduleUid,
+    rescheduleBooking,
+  });
+
+  const onMonthChange = (date: Dayjs) => {
+    setBrowsingMonthStart(date);
+    setSelectedDate(null);
+  };
+
+  const onDaySelect = (date: Dayjs) => {
+    setSelectedDate(date.format("YYYY-MM-DD"));
+  };
+
+  const onTimeSelect = (time: string) => {
+    setSelectedTimeslot(time);
+  };
+
+  useEffect(() => {
+    setLayout(isMobile ? "mobile" : "small_calendar");
+  }, [isMobile, setLayout]);
+
+  useEffect(() => {
+    if (event.isLoading) return setBookerState("loading");
+    if (!selectedDate) return setBookerState("selecting_date");
+    if (!selectedTimeslot) return setBookerState("selecting_time");
+    return setBookerState("booking");
+  }, [event, selectedDate, selectedTimeslot, setBookerState]);
+
+  return (
+    <>
+      {/* @TODO: This is imported because the DatePicker relies on it. Perhaps
+      rewrite the datepicker as well? */}
+      <CustomBranding lightVal={null} darkVal={null} />
+      {/*
+        If we would render this on mobile, it would unset the mobile variant,
+        since that's not a valid option, so it would set the layout to null.
+      */}
+      {!isMobile && (
+        <div className="dark:[&>div]:bg-darkgray-100 fixed top-2 right-3 z-10 [&>div]:bg-white">
+          <ToggleGroup
+            onValueChange={(layout) => setLayout(layout as BookerLayout)}
+            defaultValue="small_calendar"
+            options={[
+              // @TODO: Find right icon for grid, plus add a11y labels and tooltips.
+              { value: "small_calendar", label: <FiCalendar /> },
+              { value: "large_calendar", label: <FiGrid /> },
+              { value: "large_timeslots", label: <FiColumns /> },
+            ]}
+          />
+        </div>
+      )}
+      <m.div
+        layout
+        // Passing the default animation styles here as the styles, makes sure that there's no initial loading state
+        // where there's no styles applied yet (meaning there wouldn't be a grid + widths), which would cause
+        // the layout to jump around on load.
+        style={resizeAnimationConfig.small_calendar.default as MotionStyle}
+        animate={resizeAnimationConfig[layout]?.[bookerState] || resizeAnimationConfig[layout].default}
+        transition={{ ease: "easeInOut", duration: 0.4 }}
+        className={classNames(
+          "dark:bg-darkgray-100 grid w-[calc(var(--booker-meta-width)+var(--booker-main-width))] items-start overflow-x-clip bg-white [--booker-meta-width:280px] [--booker-main-width:425px] [--booker-timeslots-width:280px] md:flex-row",
+          layout === "small_calendar" &&
+            "dark:border-darkgray-300 mt-20 min-h-[450px] rounded-md border border-gray-200",
+          layout !== "small_calendar" && "h-auto min-h-screen w-screen"
+        )}>
+        <AnimatePresence>
+          <StickyOnDesktop key="meta">
+            <BookerSection area="meta">
```

Comment:
The `area` prop of the `BookerSection` component helps you place the component anywhere in the grid. If you look at the timeslots component around line 165 you also see you can pass a default and a specific grid area for a layout. This way the components can move around in grid position, based on their layout.